src/components/shared/forms/DateRange.vue

```diff
@@ -242,6 +243,9 @@
             },
 
             activate() {
+                if (this.activated) return
```

Comment:
I don't think this will work, if a component is being kept alive this will not run when it is remounted. I have updated the code sandbox to include something similar to this and it doesn't fire when the parent is remounted. Can we not use something like lodash's throttle method here, just to stop it running twice in more than 100 ms or so?

Reply:
ah good point, yes debounce seems like the best solution then.